docs/book/src/developers/releasing.md

```diff
@@ -1,18 +1,40 @@
-# Release Process
+# CAPZ Releases
 
-## Update metadata.yaml (skip for patch releases)
+## Release Cadence
+
+CAPZ minor versions (that is, 1.**5**.0 versus 1.**4**.x) are released every two months.
+
+CAPZ patch versions (for example, 1.5.**2** versus 1.5.**1**) are released every two weeks. A patch release may bypass this cadence if circumstances warrant.
+
+## Release Support
+
+The two most recent minor releases of CAPZ will be supported with bug fixes. Since minor releases arrive every two months, each minor release receives fixes for four months.
+
+For example, let's assume CAPZ v1.4.2 is the current release, and v1.3.2 is the latest in the previous minor release line. When v1.5.0 is released, it becomes the current release. v1.4.2 becomes the previous release line and remains supported. And v1.3.2 reaches end-of-life and no longer receives support through bug fixes.
+
+Note that "support" in this context refers strictly to whether or not bug fixes are backported to a release line. Please see [the support documentation](https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/SUPPORT.md) for more general information about how to get help with CAPZ.
+
+### Bug Fixes and Test Improvements
+
+Any significant user-facing bug fix that lands in the `main` branch should be backported to the current and previous release lines. Security-related fixes are automatically considered significant and user-facing.
+
+Improvements or significant changes to tests should be backported to the current release line. This is intended to minimize friction in the event of a critical test fix. Test improvements or changes may sometimes need to be backported to the previous release line in the event that tests break on all release branches.
+
```

Comment:
Let's add a discrete section to document that for experimental APIs (e.g., `AzureManagedCluster`) we may include both _features or bug fixes_ in patch releases. This an exception to the normal semver definition, but it will accelerate our effort to graduate experimental features to v1 by allowing faster feature adoption and functional iteration.

cc @CecileRobertMichon @nojnhuh @devigned @zmalik @mtougeron @LochanRn @JoelSpeed for any add'l thoughts/dissent here

Reply:
And how about another 2-3 sentences explaining that due to our minor release support contract, we want to optimize the automated cherry-pick path from main to those two minor release branches. That may include delaying codebase-wide changes like housekeeping PRs (updating downstream APIs, linting, etc) until the end of a minor release cycle. If we integrate these housekeeping PRs in the middle of a release cycle, we can then be stuck with only manual cherry-picks for the rest of the release cycle after that.

Reply:
So long as the new features are backwards compatible and don't significantly change structure or functionality I personally don't see a problem with it for experimental APIs. It really depends on what level of contract we want to provide to the users of the experimental features. I think we need to be more careful once a feature reaches beta but I definitely agree for alpha features.

Reply:
I agree with @mtougeron re: not breaking experimental APIs in a patch release. I can see where certain backwards-incompatible changes might be needed a bit sooner than the next minor release, though I think keeping experimental APIs backwards-compatible between patch releases is a good rule of thumb.

Reply:
I think this is a fair distinction:

1) additive features/functionality
2) changed/updated/removed features/functionality in response to the normal process of iterating in an experimental functional workstream

Part of our work in `AzureManagedCluster` will yield stuff that falls under #2 above. When we have a defensible reason for actually changing (i.e., breaking not just additive) an implementation classified as experimental, we should do that type of explicit breakage in a new minor release (with all-caps release notes for folks taking part in the experiment).

Does this agree with your points @nojnhuh @mtougeron?